applications/NXmx.nxdl.xml

```diff
@@ -142,6 +142,46 @@
                         <dim index="4" value="k" required="false"/>
                     </dimensions>
                 </field>
+
+                <field name="data_scaling_factor" type="NX_FLOAT" optional="true">
+                    <doc>
+                        An optional scaling factor to apply to the values in ``data``.
+
+                        The elements in ``data`` are often stored as integers for efficiency reasons and need
+                        further correction, generating floats.  The two fields data_scaling_factor and
+                        data_offset allow linear corrections using the following convention:
+
+                        .. code-block::
+
+                                corrected_data = (data + offset) * scaling_factor
+
+                        This formula will derive the corrected value, when necessary.
+
+                        data_scaling_factor is sometimes known as gain or bias and data_offset is sometimes
+                        known as pedestal or background, depending on the community.
+
+                        Use these fields to specify constants that need to be applied
```

Comment:
Does the data field's `units` attribute need to be specified in the `scaling_factor` or in `data`?

Reply:
I don't think so.  We don't specify a units attribute for data, so even though we could be converting from ADU to photons here, that wouldn't necessarily be a universal application for scaling_factor.